src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,34 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<char> inputValue = UTF8Encoding.UTF8.GetChars(value.ToArray()).AsSpan();
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json value false, Json value true, Json number.
+            if (inputValue.StartsWith("{".AsSpan(), StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("}".AsSpan(), StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"}}{Environment.NewLine}".AsSpan(), StringComparison.InvariantCultureIgnoreCase)) ||
+                inputValue.StartsWith("[".AsSpan(), StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("]".AsSpan(), StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"]{Environment.NewLine}".AsSpan(), StringComparison.InvariantCultureIgnoreCase)) ||
```

Comment:
instead of endswith `}Newline` lets start at the end and move inwards until we no longer have whitespace then use that as the last character

Reply:
I have added a trim logic and we don't need this NewLine check anymore.